Description: Fix build with bindings generated against tpm2-tss version 4
 The tss-esapi-sys crate offers both pregenerated bindings (the default) and the
 ability to request binding generation at build time. The latter is needed to
 build tss-esapi-sys (and hence sequoia-tpm) on systems for which pregenerated
 bindings are not available.

 Debian's package of the tss-esapi-sys crate is patched to automatically
 generate bindings if pre-generated bindings are not available for the
 current architecture.

 tpm2-tss version 4 increased the size of the buffer in TPM2B_PRIVATE_KEY_RSA
 from "TPM2_MAX_RSA_KEY_BYTES/2" (aka 256) to "TPM2_MAX_RSA_KEY_BYTES/2 * 5"
 https://github.com/tpm2-software/tpm2-tss/commit/e5e1a81231b1ae6b5f9201e2778a38bcd126eb40

 As a result of this is that trying to build sequoia-tpm with the bindings
 generated against tpm2-tss 4 fails with a size mismatch. This patch
 fixes the mismatch by avoiding creating a temporary array.

Author: Peter Michael Green <plugwash@debian.org>

--- rust-sequoia-tpm-0.1.0.orig/Cargo.toml
+++ rust-sequoia-tpm-0.1.0/Cargo.toml
@@ -63,6 +63,7 @@ version = "7"
 
 [dependencies.tss-esapi-sys]
 version = "0.5"
+features = ["generate-bindings"]
 
 [target."cfg(not(windows))".dev-dependencies.sequoia-openpgp]
 version = "1"
--- rust-sequoia-tpm-0.1.0.orig/src/lib.rs
+++ rust-sequoia-tpm-0.1.0/src/lib.rs
@@ -94,14 +94,10 @@ pub struct PrivateRsaKeyMaterial {
 impl From<&PrivateRsaKeyMaterial> for tss_esapi_sys::TPM2B_PRIVATE_KEY_RSA {
     fn from(private_rsa: &PrivateRsaKeyMaterial) -> Self {
         let key_prime = hex::decode(&private_rsa.prime).unwrap();
-        let mut key_prime_buffer = [0u8; 256];
-        key_prime_buffer[..key_prime.len()].clone_from_slice(&key_prime[..key_prime.len()]);
-        let key_prime_len = key_prime.len().try_into().unwrap();
-
-        Self {
-            size: key_prime_len,
-            buffer: key_prime_buffer,
-        }
+        let mut result : Self = Default::default();
+        result.buffer[..key_prime.len()].clone_from_slice(&key_prime[..key_prime.len()]);
+        result.size = key_prime.len().try_into().unwrap();
+        result
     }
 }
 
